### PR TITLE
Dead letter queue initial metrics dashboard

### PR DESCRIPTION
**Pull request by yakkomajuri**
Just building some basic scaffolding here without having given any design thought to it. 

Screenshot 2022-02-01 at 17 02 50

Management and settings tabs coming in separate PRs.

### CODE REVIEW

**Comment by yakkomajuri** [comment]
I think the scope of this PR is appropriate so will put this up for review and handle other things in subsequent PRs.

**Comment by yakkomajuri** [comment]
Updated UI for single value metrics:

Screenshot 2022-02-02 at 09 07 21

File: frontend/src/scenes/instance/DeadLetterQueue/DeadLetterQueue.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+const { TabPane } = Tabs
+
+export function DeadLetterQueue(): JSX.Element {
+    const { user } = useValues(userLogic)
+    const { activeTab } = useValues(deadLetterQueueLogic)
+    const { setActiveTab } = useActions(deadLetterQueueLogic)
+
+    if (!user?.is_staff) {
+        return <></>
```
Comment (tiina303) [inline frontend/src/scenes/instance/DeadLetterQueue/DeadLetterQueue.tsx:23]: Don't we have some default "You need to be staff to see this page" + link to docs on how to get to staff status?

File: frontend/src/scenes/instance/DeadLetterQueue/MetricsTab.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    const { singleValueMetrics, tableMetrics, deadLetterQueueMetricsLoading } = useValues(deadLetterQueueLogic)
+
+    if (!user?.is_staff) {
+        return <></>
+    }
+
+    return (
+        <div>
+            <br />
+
```
Comment (tiina303) [inline frontend/src/scenes/instance/DeadLetterQueue/MetricsTab.tsx:20]: maybe add a refresh button?

File: frontend/src/scenes/instance/DeadLetterQueue/deadLetterQueueLogic.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }),
+
+    loaders: () => ({
+        deadLetterQueueMetrics: [
+            [] as SystemStatusRow[],
+            {
+                loadDeadLetterQueueMetrics: async () => {
+                    if (!userLogic.values.user?.is_staff) {
+                        return []
+                    }
```
Comment (tiina303) [inline frontend/src/scenes/instance/DeadLetterQueue/deadLetterQueueLogic.ts:38]: Do we need this, feels repetitive, the API already has that restriction?

File: posthog/api/dead_letter_queue.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            {"key": "dlq_size", "metric": "Total events in dead letter queue", "value": get_dead_letter_queue_size()}
+        )
+
+        metrics.append(
+            {
+                "key": "dlq_events_last_24h",
+                "metric": "Events sent to dead letter queue in the last 24h",
+                "value": get_dead_letter_queue_size(),
+            }
+        )
```
Comment (tiina303) [inline posthog/api/dead_letter_queue.py:33]: not used and `get_dead_letter_queue_size` doesn't look correct either

File: posthog/api/dead_letter_queue.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+        return Response({"results": {"overview": metrics}})
+
+
+def get_dead_letter_queue_size() -> int:
+    return sync_execute("SELECT count(*) FROM events_dead_letter_queue")[0][0]
+
+
+def get_dead_letter_queue_events_last_day() -> int:
```
Comment (tiina303) [inline posthog/api/dead_letter_queue.py:72]: nit for clarity
```suggestion
def get_dead_letter_queue_events_last_24h() -> int:
```

File: posthog/api/dead_letter_queue.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    return sync_execute("SELECT count(*) FROM events_dead_letter_queue WHERE _timestamp >= (NOW() - INTERVAL 1 DAY)")[
```
Comment (tiina303) [inline posthog/api/dead_letter_queue.py:73]: should we use `error_timestamp` here - could look inconsistent with the by day table?

File: posthog/api/dead_letter_queue.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        0
+    ][0]
+
+
+def get_dead_letter_queue_events_per_error() -> List[Union[str, int]]:
+    return sync_execute("SELECT error, count(*) AS c FROM events_dead_letter_queue GROUP BY error ORDER BY c DESC")
```
Comment (tiina303) [inline posthog/api/dead_letter_queue.py:79]: Let's also query and add the latest error timestamp (here & UI), I find that nice to have in sentry, especially sorting the table by that column.
Do we need an order by here (faster without and UI might impose a different sorting)?
Same for the next two.

```suggestion
    return sync_execute("SELECT error, count(*) AS c, max(error_timestamp) FROM events_dead_letter_queue GROUP BY error")
```

**Review by tiina303 — APPROVED** [review]
🙌

File: frontend/src/scenes/instance/DeadLetterQueue/deadLetterQueueLogic.ts (hunk 3 above)
Comment (yakkomajuri) [reply frontend/src/scenes/instance/DeadLetterQueue/deadLetterQueueLogic.ts:38]: Why hit the endpoint unnecessarily though

File: posthog/api/dead_letter_queue.py (hunk 4 above)
Comment (yakkomajuri) [reply posthog/api/dead_letter_queue.py:33]: ah yup typo from ctrl c ctrl v

File: frontend/src/scenes/instance/DeadLetterQueue/deadLetterQueueLogic.ts (hunk 3 above)
Comment (tiina303) [reply frontend/src/scenes/instance/DeadLetterQueue/deadLetterQueueLogic.ts:38]: I assume we'd never reach this, making sure we load it only if needed, which would be only for staff users

File: posthog/api/dead_letter_queue.py (hunk 7 above)
Comment (tiina303) [reply posthog/api/dead_letter_queue.py:95]: What were your thoughts here? Specifically I find the total events per error not super useful longer term if we don't have last error timestamp

Comment (yakkomajuri) [reply posthog/api/dead_letter_queue.py:95]: I just merged as is to iterate. The frontend currently expects 2 columns only, so would need to do some refactoring there.

**Pull request merged**